src/components/camera/client.cpp

```diff
@@ -72,6 +74,10 @@ Camera::properties CameraClient::get_properties() {
 };
 
 CameraClient::CameraClient(std::string name, std::shared_ptr<grpc::Channel> channel_)
-    : channel_(channel_), stub_(viam::component::camera::v1::CameraService::NewStub(channel_)){};
+    : channel_(channel_), stub_(viam::component::camera::v1::CameraService::NewStub(channel_)) {
+    name_ = std::move(name);
```

Comment:
So I was unable to initialize `name_` in the initializer list because it's not a part of `CameraClient`, it's part of `ResourceBase`. Seems like there's a workaround in this [SO](https://stackoverflow.com/questions/2290733/initialize-parents-protected-members-with-initialization-list-c) post, but it seemed not worth it to me because it necessitates several chains of `protected` constructors (`ResourceBase` -> `ComponentBase` -> `Camera` -> CameraClient`)